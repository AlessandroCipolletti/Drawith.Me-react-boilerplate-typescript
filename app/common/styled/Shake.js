import styled, { css, keyframes } from 'styled-components'

const maxPx = 4

const shakeAnimation = keyframes`
  10%, 90% {
    transform: translate3d(${-maxPx / 4}px, 0, 0);
  }
  20%, 80% {
    transform: translate3d(${maxPx / 2}px, 0, 0);
  }
  30%, 50%, 70% {
    transform: translate3d(${-maxPx}px, 0, 0);
  }
  40%, 60% {
    transform: translate3d(${maxPx}px, 0, 0);
  }
`

const Shake = styled.div`
  animation: ${props => props.isShaking ? css`${shakeAnimation} ${props.theme.timing.shakeAnimation} ease-in-out 0s infinite` : ''};
  perspective: 1000px;
`

export default Shake
