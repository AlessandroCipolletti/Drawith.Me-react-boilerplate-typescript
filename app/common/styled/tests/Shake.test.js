import React from 'react'
import { render } from 'react-testing-library'

// Import de Theme Provider
import { ThemeProvider } from 'styled-components'
import Theme from 'common/theme'

import Shake from '../Shake'

describe('<Shake />', () => {
  const renderComponent = (props = {}) =>
    render(
      <ThemeProvider theme={Theme}>
        <Shake {...props}>
          Content
        </Shake>
      </ThemeProvider>,
    )

  it('Should render and match the snapshot while shaking', () => {
    const {
      container: { firstChild },
    } = renderComponent({ isShaking: true })

    expect(firstChild).toMatchSnapshot()
  })

  it('Should render and match the snapshot while not shaking', () => {
    const {
      container: { firstChild },
    } = renderComponent({ isShaking: false })

    expect(firstChild).toMatchSnapshot()
  })
})
