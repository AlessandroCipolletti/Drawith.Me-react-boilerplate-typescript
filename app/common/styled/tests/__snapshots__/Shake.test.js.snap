// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shake /> Should render and match the snapshot while not shaking 1`] = `
<div
  class="Shake-e77na9-0 iHCLPl"
>
  Content
</div>
`;

exports[`<Shake /> Should render and match the snapshot while shaking 1`] = `
<div
  class="Shake-e77na9-0 ebpSXC"
>
  Content
</div>
`;
